### ISSUE TITLE
Simple Form

### ISSUE THREAD

**Issue by drujensen** [johnjansen/crystal-libraries-needed#10]

**Comment by molovo** [comment]
@drujensen What did you mean by this? A form builder, or something different?

I've built a couple of those in PHP for different projects, with an API very similar to Laravel's, so if that's the case I'd happily add it to my to-do list.

**Comment by bigtunacan** [comment]
@molovo Yes; that is what he is referring to.  You can see Simple Form here.  https://github.com/plataformatec/simple_form 

It's the top form gem in Ruby.  For a long time Formtastic was the most used, but it became unmaintained for a long time. (https://github.com/justinfrench/formtastic)  Although now that gem is once again up to date and maintained as well.

**Comment by felipeelias** [comment]
@molovo have you started working on this? I'm willing to give it a try if you haven't

**Comment by molovo** [comment]
@felipeelias I haven't, and I'm unlikely too any time soon, so crack on

**Comment by molovo** [comment]
Is there a decent data validation library out there that we could integrate with this?

**Comment by felipeelias** [comment]
@molovo not that I know of. My idea is that the form builder may expect some object that defines a simple interface, like `valid?` and `errors` to display validation messages.

I didn't get there yet but I have started something [here](https://github.com/felipeelias/forms.cr). It's very minor but I'm already using in one project of mine.
